pkg/controller/infrastructure/infraflow/delete.go

```diff
@@ -36,6 +36,8 @@ func (fctx *FlowContext) buildDeleteGraph() *flow.Graph {
 
 	needToDeleteNetwork := fctx.config.Networks.ID == nil
 	needToDeleteRouter := fctx.config.Networks.Router == nil
+	// skip deletion of the subnet if we need to delete the network (it will be deleted anyway), or if the subnet is user provided.
```

Comment:
```suggestion
	// skip deletion of the subnet if it is user provided.
```
I've moved the other part of the comment closer to the relevant line (see below)